test/types/records/generic/genericRecordInFunction.chpl

```diff
@@ -0,0 +1,6 @@
+proc foo(args) {
```

Comment:
I would add a comment about the purpose behind adding the test - it should mention that the program previously encountered an internal error.  That'll make it less likely that someone adjusts the test in the future to not exercise what it was originally exercising